services/user.service.js

```diff
@@ -1 +1,108 @@
-// put api for user
+const MealPlan = require("../models/meal-plan.model");
+const { NotFoundError, UnauthorizedError } = require("../helpers/errorHandlers");
+const User = require("../models/user.model");
+const jwt = require("jsonwebtoken");
+const { authToken } = require("../middleware/auth");
+
+exports.getUserMealPlans = async (userId) => {
+  const mealPlan = await MealPlan.find({ author: userId });
+  if (!mealPlan) {
+    throw new NotFoundError("Meal plans don't exist");
+  }
+  return mealPlan;
+};
+
+exports.getUserMealPlan = async (id) => {
+  const mealPlan = await MealPlan.findById({ _id: id });
+  if (!mealPlan) {
+    throw new NotFoundError("Meal plan doesn't exist");
+  }
+  return mealPlan;
+};
+
+exports.udpateUserMealPlan = async (id, reqBody) => {
+  const mealPlan = await MealPlan.findById({ _id: id });
+  if (!mealPlan) {
+    throw new NotFoundError("Meal plan doesn't exist");
+  }
+  await MealPlan.findByIdAndUpdate({ _id: id }, reqBody, {
+    new: true,
+  });
+  return true;
+};
+
+exports.deleteUserMealPlan = async (id) => {
+  const mealPlan = await MealPlan.findById({ _id: id });
+  if (!mealPlan) {
+    throw new NotFoundError("Meal plan doesn't exist");
+  }
+  await MealPlan.deleteOne({ _id: id });
+  return true;
+};
+
+exports.createUserMealPlan = async (reqBody) => {
+  const mealPlan = new MealPlan({
```

Comment:
you can just 

new MealPlan({...body})

Reply:
it's not my code